### PR TITLE
workflow mode on client

### PR DESCRIPTION
**Pull request by Yancey0623**
Example SQL program:
``` sql
%%sqlflow
select *
from iris.train
limit 5;

SELECT *
FROM iris.train
TO TRAIN DNNClassifier
WITH
  model.n_classes = 3,
  model.hidden_units = [10, 10],
  train.epoch = 100
COLUMN sepal_length, sepal_width, petal_length, petal_width
LABEL class
INTO sqlflow_models.my_dnn_model;
```

output logs:

``` text
sqlflow> select *
from iris.train
limit 5;
+--------------+-------------+--------------+-------------+-------+
| SEPAL LENGTH | SEPAL WIDTH | PETAL LENGTH | PETAL WIDTH | CLASS |
+--------------+-------------+--------------+-------------+-------+
|          6.4 |         2.8 |          5.6 |         2.2 |     2 |
|            5 |         2.3 |          3.3 |           1 |     1 |
|          4.9 |         2.5 |          4.5 |         1.7 |     2 |
|          4.9 |         3.1 |          1.5 |         0.1 |     0 |
|          5.7 |         3.8 |          1.7 |         0.3 |     0 |
+--------------+-------------+--------------+-------------+-------+

sqlflow> SELECT *
FROM iris.train
TO TRAIN DNNClassifier
WITH
model.n_classes = 3,
model.hidden_units = [10, 10],
train.epoch = 100
COLUMN sepal_length, sepal_width, petal_length, petal_width
LABEL class
INTO sqlflow_models.my_dnn_model;
2019-12-25 16:21:23.409781: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2

2019-12-25 16:21:23.541065: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2207995000 Hz

2019-12-25 16:21:23.543140: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5635d6dcb9d0 executing computations on platform Host. Devices:

```

### CODE REVIEW

File: sqlflow/client.py [hunk 1]
```diff
@@ -180,4 +207,6 @@ def _decode_any(cls, any_message):
                 any_message.Unpack(timestamp_message)
                 return timestamp_message.ToDatetime()
             raise TypeError("Unsupported type {}".format(any_message))
-
+    
+    def _enable_argo_mode(self):
```
Comment (tonyyang-svail) [inline sqlflow/client.py]: Can we determine the mode based on the return value of `Run` RPC call? For example, if the return value contains a job message, the client should execute fetch. This ease the deployment by avoiding configuring environment variable `SQLFLOW_ARGO_MODE`.

**Review by tonyyang-svail — COMMENTED** [review]
Where are the tests for `Fetch`? :)

File: proto/sqlflow/proto/sqlflow.proto [hunk 2]
```diff
@@ -10,13 +10,65 @@ service SQLFlow {
     // SQL statements like `SELECT ...`, `DESCRIBE ...` returns a rowset.
     // The rowset might be big. In such cases, Query returns a stream
     // of RunResponse
+    // 
+    // SQLFlow implements the Run interface with two mode:
     //
+    // 1. Local model
+    // The SQLFlow server execute the SQL statements on the local host.
```
Comment (typhoonzero) [inline proto/sqlflow/proto/sqlflow.proto]: local host -> localhost.

**Review by typhoonzero — DISMISSED** [review]

File: sqlflow/client.py [hunk 3]
```diff
@@ -114,54 +128,59 @@ def execute(self, operation):
...
-                            eoe = res
-                            break
-                        _LOGGER.debug(res.message.message)
-                        all_messages.append(res.message.message)
-                    compound_message.add_message('\n'.join(all_messages), eoe)
+                    _LOGGER.info(first.message.message)
+                    for res in reader.read():
+                        _LOGGER.info(res.message.message)
+            elif oneof_first == 'job':
+                self.read_fetch_response(first.job.id)
```
Comment (typhoonzero) [inline sqlflow/client.py]: Does this line duplicate with line 178?

Comment (Yancey0623) [reply sqlflow/client.py]: No, the first response iteration may be `message` type which contains job logs, and the last response is `job` type.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**